opensrp-reveal/src/main/java/org/smartregister/reveal/util/RevealMapHelper.java

```diff
@@ -53,16 +86,85 @@ public static void addSymbolLayers(@NonNull Style mMapboxMapStyle, Context conte
         symbolLayer.setFilter(eq(get(TYPE), StructureType.MOSQUITO_COLLECTION_POINT));
         mMapboxMapStyle.addLayer(symbolLayer);
 
-
+        // larval breeding symbol layer
         icon = BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_breeding);
         mMapboxMapStyle.addImage(LARVAL_BREEDING_ICON, icon);
         symbolLayer = new SymbolLayer(LARVAL_BREEDING_LAYER, context.getString(R.string.reveal_datasource_name));
         symbolLayer.setProperties(
                 iconImage(LARVAL_BREEDING_ICON),
                 iconSize(dynamicIconSize));
-
         symbolLayer.setFilter(eq(get(TYPE), StructureType.LARVAL_BREEDING_SITE));
-
         mMapboxMapStyle.addLayer(symbolLayer);
     }
+
+    public static void addCaseIndexBoundary(MapboxMap mapboxMap, Context context) {
+        Style mMapboxMapStyle = mapboxMap.getStyle();
+        try {
+            Expression dynamicIconSize = interpolate(linear(), zoom(),
+                    literal(13.98f), literal(1),
+                    literal(17.79f), literal(3f),
+                    literal(18.8f), literal(4));
+
+            // index case symbol layer
+            if (getInterventionLabel() == R.string.focus_investigation) {
+                Bitmap icon = BitmapFactory.decodeResource(context.getResources(), R.drawable.ic_index_case_target_icon);
+                mMapboxMapStyle.addImage(INDEX_CASE_TARGET_ICON, icon);
+                SymbolLayer symbolLayer = new SymbolLayer(INDEX_CASE_SYMBOL_LAYER, context.getString(R.string.reveal_datasource_name));
+                symbolLayer.setProperties(
+                        iconImage(INDEX_CASE_TARGET_ICON),
+                        iconSize(dynamicIconSize));
+                symbolLayer.setFilter(eq(get(IS_INDEX_CASE), true));
+                if (mMapboxMapStyle.getLayer(INDEX_CASE_SYMBOL_LAYER) == null) {
+                    mMapboxMapStyle.addLayer(symbolLayer);
+                }
+            }
+
+            // index case circle layer
+            GeoJsonSource indexCaseSource = new GeoJsonSource(INDEX_CASE_SOURCE);
```

Comment:
This method needs to be on its own method, the lines upto 118, can be called only once in the app.

From here you can break down so that you two methods for Creating index case circle layer and another for updating it in case it already exists

Reply:
Could you rephrase what the issue is here?

Reply:
You need to separate Creating Layers from Creating the Sources. Layers should be created only once. However for the sources it be possible to create a method that can update the source.

Reply:
Done.